cmake/FindSDL2.cmake

```diff
@@ -29,6 +28,8 @@ if(NOT SDL2_FOUND)
     endif()
 endif()
 
+find_library(SDL2main_LIBRARY SDL2main)
```

Comment:
We don't use `SDL2main_LIBRARY`, so perhaps all logic related to `SDL2main` can be removed from this cmake script?